dice.py

```diff
@@ -1,23 +1,88 @@
+"""
+
 import random
 
-repeat = True
-while repeat:
-    no = random.randint(1, 6)
-    print("You rolled", no)
-    if no == 1:
-        print("You feel somewhat disappointed")
-    if no == 2:
-        print("You look at the two dots and begin to mourn a lover lost at sea")
-    if no == 3:
-        print("Three is the magic number, you turn into a frog")
-    if no == 4:
-        print("4! You win...not really, you're just rolling a dice. Get some mates!")
-    if no == 5:
-        print("You bleed a little")
-    if no == 6:
-        print("You begin to wonder what this point of this is")
-    if no == 7:
-        print("err...what? This is literally impossible")
-
-    print("Do you want to roll again Y/N (please roll again, it...it pleases me?)")
-    repeat = "Y" in input()
\ No newline at end of file
+"""Configurables"""
+number_of_dice = 3
+faces = 6
+
+
+def main():
+    repeat = True
+    while repeat:
+        display(roll(number_of_dice, faces))
+        print("Do you want to roll again Y/N (please roll again, it...it pleases me?)")
+        repeat = "Y" in input().upper()
+
+
+def roll(number_of_dice, faces):
+    return [random.randint(1, faces) for n in range(number_of_dice)]
+
+
+def display(dice):
+    if all(number <= 6 for number in dice):
+        for number in dice:
+            if number == 1:
+                print(" -------")
+                print("｜     ｜")
+                print("｜  o  ｜")
+                print("｜     ｜")
+                print(" -------")
+            elif number == 2:
+                print(" -------")
+                print("｜o    ｜")
+                print("｜     ｜")
+                print("｜    o｜")
+                print(" -------")
+            elif number == 3:
+                print(" -------")
+                print("｜o    ｜")
+                print("｜  o  ｜")
+                print("｜    o｜")
+                print(" -------")
+            elif number == 4:
+                print(" -------")
+                print("｜o   o｜")
+                print("｜     ｜")
+                print("｜o   o｜")
+                print(" -------")
+            elif number == 5:
+                print(" -------")
+                print("｜o   o｜")
+                print("｜  o  ｜")
+                print("｜o   o｜")
+                print(" -------")
+            elif number == 6:
+                print(" -------")
+                print("｜o   o｜")
+                print("｜o   o｜")
+                print("｜o   o｜")
+                print(" -------")
+    print("You rolled " + return_dice_as_text(dice))
+
+
+def return_dice_as_text(dice):
+    output = ""
+    for i in range(len(dice)):
+        if i == 0:
+            output += f"a {dice[i]}"
+            if len(dice) == 1:
+                break
+        elif i == len(dice) - 1:
+            output += f" and a {dice[i]}; totaling {sum(dice)}"
+        else:
+            output += f", a {dice[i]}"
+    return output
+
+
+if __name__ == "__main__":
```

Comment:
What does this do?

Reply:
Very good question. Using this is very good practice and you should do it for anything more complicated than a short little script. If you want to use this dice script for anything else then you'll need to import it, just like you did with random. When you import a script or any functions from a script then that script gets run. This is necessary to make sure everything is setup correctly. When the script is run it gets given a name (with the variable `__name__`). If you were to run this script from the outside (ie as an import) then it would get called `dice`. If, however, you run this script from the inside (like you've been doing) then it gets the name `__main__`. So this means that all the dice stuff only happens if you want to run the script as a script and not when you import it into a different program. That might not be the best explanation so maybe you should google this. It's something employers would be looking for and is very important for larger projects.